src/Config/SystemConfigManager.ts

```diff
@@ -37,6 +38,41 @@ export class SystemConfigReader {
     };
   }
 
+  public enableBackgroundRotation(onRotation: (config: SystemConfig) => void) {
+    this.isBackgroundRotationEnabled = true;
+
+    (async () => {
```

Comment:
Not sure how floating promises can replace IIFE. Could you please clarify a little bit?

If you mean, that `enableBackgroundRotation` can return a promise, than I think a client of this method should not care about handling "background process" promise handling, since the method just enables it - the rest happens in background

Reply:
Well, extract this nested function into a separate method that returns a Promise, and then delegate the call to it from `enableBackgroundRotation` without awaiting the result. The public interface should remain the same.

Reply:
> Note: IIFE is old fashioned, and not readable due to the complexity

Thanks, good point.

Extracted "run" logic to `runBackgroundRotation`